File: Assignment 01/Program 05/prog5.py
#Write a C program to find the maximum element in a 2D array in using dynamic memory allocation

import numpy as np

row = int(input("Enter the number of rows : "))
col = int(input("Enter the number of columns: "))

a = list(map(int,input("Enter the elements: ").split()))
mat1 = np.array(a)
mat2 = mat1.reshape(row,col)
print(mat2)

max = 0
for i in range(row):
    for j in range(col):
        if mat2[i][j] > max:
            max = mat2[i][j]

print("The maximum element :",max)

File: Assignment 01/Program 05/prog5.c
//Write a C program to find the maximum element in a 2D array in using dynamic memory allocation

#include<stdio.h>
#include<stdlib.h>
#include<limits.h>

int main(){
    int row,col;
    printf("Enter the no. of row and column respectively: ");
    scanf("%d %d",&row,&col);
    
    int **arr = (int**)malloc(row*sizeof(int*));

    for (int i = 0; i < row; i++)
    {
        arr[i] = (int*)malloc(col*sizeof(int));
    }
    
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            printf("Enter the element [%d][%d]:",i+1,j+1);
            scanf("%d",&arr[i][j]);
        }
    }

    int max = INT_MIN;
    for (int i = 0; i < row; i++)
    {
        for (int j = 0; j < col; j++)
        {
            if(arr[i][j]> max)
                max = arr[i][j];
        }
        
    }
    printf("The maximum element is: %d",max);
    free(arr);
    return 0;
}